metaflow/plugins/kfp/kfp_s3_sensor.py

```diff
@@ -0,0 +1,58 @@
+"""
+This function is called within the s3_sensor_op running container.
+(1) It decodes the formatter function's code and runs it to obtain a final S3 path
+(2) It splits the formatted path into an S3 bucket and key 
+(3) It polls for an object with the specified bucket and key until timeout
+"""
+
+
+def wait_for_s3_path(
+    path: str,
+    timeout_seconds: int,
+    polling_interval_seconds: int,
+    func_code_encoded: str,
+    flow_parameters_json: str,
+) -> None:
+    import boto3
+    import botocore
+    import base64
+    import marshal
+    import json
+    import time
+    from datetime import date
+    from typing import Tuple
+
+    def split_s3_path(path: str) -> Tuple[str, str]:
+        path = path.replace("s3://", "")
```

Comment:
what if this were `s3a`?

Reply:
Refer to https://github.com/zillow/metaflow/blob/feature/kfp/metaflow/datatools/s3.py#L216 for example. `urlparse` may be handy.

Reply:
is there a boto3 function to parse the s3 path?

Reply:
https://stackoverflow.com/questions/42641315/s3-urls-get-bucket-name-and-path